### PR TITLE
adds remove function for cache

### PR DESCRIPTION
**Pull request by dylanlott**

### CODE REVIEW

File: pkg/overlay/cache.go [hunk 1]
```diff
@@ -128,3 +131,18 @@ func (o *Cache) Put(ctx context.Context, nodeID storj.NodeID, value pb.Node) err
 
 	return o.DB.Put(nodeID.Bytes(), data)
 }
+
+// Delete will remove the node from the cache. Used when a node hard disconnects or fails
+// to pass a PING multiple times.
+func (o *Cache) Delete(ctx context.Context, id storj.NodeID) error {
+	if id.IsZero() {
+		return ErrDelete
```
Comment (mniewrzal) [inline pkg/overlay/cache.go]: Shouldn't this be `ErrEmptyNode`? Like in Get method.

File: pkg/overlay/cache_test.go [hunk 2]
```diff
@@ -98,6 +98,13 @@ func testCache(ctx context.Context, t *testing.T, store storage.KeyValueStore, s
 			assert.Error(t, err)
 		}
 	}
+
+	{ // Delete
+		err := cache.Delete(ctx, valid1ID)
+		assert.NoError(t, err)
+		err = cache.Delete(ctx, storj.NodeID{})
+		assert.Error(t, err)
```
Comment (mniewrzal) [inline pkg/overlay/cache_test.go]: Maybe also testing invalid node id?

File: pkg/overlay/cache.go [hunk 3]
```diff
@@ -21,6 +21,9 @@ const (
 	OverlayBucket = "overlay"
 )
 
+// ErrDelete is returned when there was an error deleting something
+var ErrDelete = errs.New("error deleting key:")
```
Comment (mniewrzal) [inline pkg/overlay/cache.go]: Why `:` here?

File: pkg/overlay/cache.go [hunk 4]
```diff
@@ -128,3 +131,18 @@ func (o *Cache) Put(ctx context.Context, nodeID storj.NodeID, value pb.Node) err
 
 	return o.DB.Put(nodeID.Bytes(), data)
 }
+
+// Delete will remove the node from the cache. Used when a node hard disconnects or fails
+// to pass a PING multiple times.
+func (o *Cache) Delete(ctx context.Context, id storj.NodeID) error {
```
Comment (egonelbre) [inline pkg/overlay/cache.go]: Why would we want to remove things from cache? I mean we can flag it as problematic or something instead.

File: pkg/overlay/cache.go [hunk 5]
```diff
@@ -128,3 +131,18 @@ func (o *Cache) Put(ctx context.Context, nodeID storj.NodeID, value pb.Node) err
...
+	if id.IsZero() {
+		return ErrDelete
+	}
+
+	err := o.DB.Delete(id.Bytes())
+	if err != nil {
+		return ErrDelete
```
Comment (egonelbre) [inline pkg/overlay/cache.go]: no need for the separate error class

File: pkg/overlay/cache.go (hunk 4 above)
Comment (coyle) [reply pkg/overlay/cache.go]: It's assumed that things in the cache are good to go. If we have a bad node in the cache it should be removed.

Comment (egonelbre) [reply pkg/overlay/cache.go]: Hard disconnect doesn't mean it's a bad node. E.g. losing network for half an hour would mean that the whole cache could be wiped.

Comment (coyle) [reply pkg/overlay/cache.go]: and then the cache would rebuild itself the next run. The assumption is everything in the cache was contacted successfully within the last hour. It's not a bad node if it's removed, it's a node we didn't successfully speak with.

Comment (dylanlott) [reply pkg/overlay/cache.go]: To add on to this: There are definitely cases where we want to remove a node from the cache, but also we aren't going to remove them after a single failure to communicate. I think we had talked in the past about removing nodes after X failed attempts, or X amount of time being unresponsive. This is just adding the method to the cache for that removal.

File: pkg/overlay/cache.go (hunk 3 above)
Comment (dylanlott) [reply pkg/overlay/cache.go]: Fixed

File: pkg/overlay/cache_test.go (hunk 2 above)
Comment (dylanlott) [reply pkg/overlay/cache_test.go]: Added another test

File: pkg/overlay/cache.go (hunk 1 above)
Comment (dylanlott) [reply pkg/overlay/cache.go]: Fixed

File: pkg/overlay/cache_test.go [hunk 6]
```diff
@@ -98,6 +98,20 @@ func testCache(ctx context.Context, t *testing.T, store storage.KeyValueStore, s
 			assert.Error(t, err)
 		}
 	}
+
+	{ // Delete
+		// Test standard delete
+		err := cache.Delete(ctx, valid1ID)
```
Comment (mobyvb) [inline pkg/overlay/cache_test.go]: nit - also test that `cache.Get` returns an error for `valid1ID` after this assertion

Comment (mobyvb) [reply pkg/overlay/cache_test.go]: and maybe that it does not return an error before this assertion

Comment (dylanlott) [reply pkg/overlay/cache_test.go]: Test added 👍

**Pull request closed without merging**